src/components/vm/disks/diskEdit.tsx

```diff
@@ -203,20 +284,21 @@ export class EditDiskModal extends React.Component {
         const Dialogs = this.context;
         const { vm, disk, idPrefix, supportedDiskBusTypes } = this.props;
 
+        const disk2 = vm.disks[disk.target]; // XXX - do we always have disk === disk2?
+
         const defaultBody = (
             <Form isHorizontal>
                 <NameRow idPrefix={idPrefix}
-                         diskType={vm.disks[disk.target].type}
-                         name={getDiskFullName(vm.disks[disk.target])} />
+                         diskType={disk2.type || ""}
+                         name={getDiskFullName(disk2)} />
 
                 <AccessRow dialogValues={this.state}
-                           diskDevice={disk.device}
+                           diskDevice={disk.device || "disk"}
```

Comment:
This also changes runtime behavior (same as a couple of lines down). Thus this should either become its own commit with justification, or another cockpit.assert that `disk.device` is defined.